### ISSUE TITLE
BUG cant connect bitbox02

### ISSUE THREAD

**Issue by vogues** [RabbyHub/Rabby#895]
please check connection w your wallet error when trying to link my bitbox hardware wallet.

on brave, newest bitbox02 firmware. this was not an issue in the past.

there is a connection to the hardware wallet and bitbox app on the computer. the display is on and signed in. everything works from a hardware connection perspective. the only time i run into an issue is linking a new address via rabby.

it is also possible to connect bitbox via MEW, so it appears it is not a hardware issue.

**Comment by vvvvvv1vvvvvv** [comment]
@benma Could you help on this issue?

**Comment by benma** [comment]
Does it work on Chrome?

Is your BitBoxBridge service running? Check the output of http://127.0.0.1:8178/.

Try disabling the Brave shield, which blocks connections to the BitBoxBridge.

**Comment by vogues** [comment]
chrome doesnt work after more testing.

**Comment by jadzeidan** [comment]
I'm running the latest firmware, 9.12.0 and the latest Rabby release, 0.41.1

Do you have another computer you can try on?

Maybe try reinstalling the Bridge as well?

**Comment by benma** [comment]
@vogues please try the suggestions of @jadzeidan.

You could also try setting it up on a new Chrome/Brave user profile, to see if the Brave data folder/cache might be corrupt. If it works with a clean slate, it indicates a bug in Rabby.

It would also be interesting to see if the console shows anything useful. In the extensions manager, you can inspect the views (background.html and maybe one more) and see the console output. Maybe it outputs something useful when attempting to connect.

**Comment by vogues** [comment]
> @vogues please try the suggestions of @jadzeidan.
> 
> You could also try setting it up on a new Chrome/Brave user profile, to see if the Brave data folder/cache might be corrupt. If it works with a clean slate, it indicates a bug in Rabby.
> 
> It would also be interesting to see if the console shows anything useful. In the extensions manager, you can inspect the views (background.html and maybe one more) and see the console output. Maybe it outputs something useful when attempting to connect.

even with a new profile it is still being blocked.

may i have more specific instructions to inspect the views please

**Comment by vogues** [comment]
is it possible my bitbox is outputting to a different port?

**Comment by stas** [comment]
@vogues make sure there are no other apps using the device in the same time (that includes the official BitBox app).

The instructions should guide the user to do that btw.

**Comment by vogues** [comment]
> . the error is still "please check connection

I have gone and replaced my original bitbox02- I am still unable to connect to rabby

**Comment by vogues** [comment]
> @vogues make sure there are no other apps using the device in the same time (that includes the official BitBox app).
> 
> The instructions should guide the user to do that btw.

Rabby will not connect to bitbox on either browser with a brand new hardware device with everything updated. I've tried with and without the bridge on. I've also tried it with the bitbox hardware locked and unlocked.

Why can't it get access?

**Comment by vogues** [comment]
@stas @jadzeidan @benma

Rabby will not connect to bitbox on either chrome or brave browser with a brand new hardware device with everything updated. I've tried with and without the bridge on. I've also tried it with the bitbox hardware locked and unlocked.

This can't just be isolated to me can it?

**Comment by vogues** [comment]
@jadzeidan 

it will not work on the new computer either

**Comment by stas** [comment]
@vogues I'm not sure I understood your response here:
https://github.com/RabbyHub/Rabby/issues/895#issuecomment-1274431682

The idea is that the bridge seems to handle just one client at a time (be that Rabby, or the BitBox app). Could you confirm that there are no other wallets/apps using the device? One you unlocked the device, just close the BitBox app.

**Comment by vogues** [comment]
@stas 

I can confirm nothing was using rabby on both computers, and that it was unlocked.

**Comment by vogues** [comment]
The second computer I did a fresh install, so it was clear of any third-party software potentially using the device

**Comment by jadzeidan** [comment]
@vogues what operating system are you using for both computers you tried on?

**Comment by vogues** [comment]
@jadzeidan 

One windows 10, the other 11.

**Comment by jadzeidan** [comment]
@vogues Can you please verify that the BitBoxBridge is running in "Services". You can do this by

1. Run "services"

![image](https://user-images.githubusercontent.com/42831937/208251215-ab32597c-ed25-4451-b8dd-33f706890916.png)

2. In the "Services" window, check that the "BitBoxBridge" status is "Running":

![image](https://user-images.githubusercontent.com/42831937/208251273-e93cfe70-53ab-4cab-990e-6ddca182ea5e.png)

**Comment by vogues** [comment]
@jadzeidan 
I thought you didn't want it running. It shuts down when the bridge is closed.

At any rate, I have tried it with that service running and not running.

**Comment by jadzeidan** [comment]
@vogues 

I think you might be confusing the BitBoxBridge with the BitBoxApp?

The BitBoxBridge needs to be running to use Rabby. It is a background application so the BitBox02 can connect to web wallets.

The BitBoxApp should **not** be running.

Please confirm you have installed the [BitBoxBridge](https://shiftcrypto.ch/download/#bridge) and it is running in the **Services** application like in the screenshot above.

**Comment by vogues** [comment]
@jadzeidan 

Oh interesting. Looks like there is an overlap in the names of services which is why I thought the bridge was running. At any rate you have solved the issue- I downloaded the bridge separately and it worked.

Thank you!

**Comment by stas** [comment]
Lol, @jadzeidan btw, is there a reason why the bridge and the app are not bundled?

Tbh, I'm on the side of @vogues here, this is confusing for regular end-user :see_no_evil:

**Comment by pozylon** [comment]
This is definitely an issue, I have it too and I couldn't solve it.

Others have issues too, see https://github.com/digitalbitbox/bitbox-bridge/issues/43 for example. And it's most propably not a problem of the bridge because I can connect to the other supported wallets of bitbox just fine.

I also tried it with an old version of chrome, chomium, brave and edge and I tried various versions of Rabby between 44 and the current release all without luck.

I'd hunt this down and submit a PR but sadly, I found out that Rabby is not quite open-source, there is private packages missing to run it in dev mode, see 
https://github.com/RabbyHub/Rabby/issues/1317
https://github.com/RabbyHub/Rabby/issues/1163

If anybody could unlock me in some ways that be great.
